tapfreighter/parcel.go

```diff
@@ -286,7 +286,101 @@ func (p *PreSignedParcel) kit() *parcelKit {
 
 // Validate validates the parcel.
 func (p *PreSignedParcel) Validate() error {
-	// TODO(ffranr): Add validation where appropriate.
+	if p.vPkt == nil {
+		return fmt.Errorf("no virtual transaction in pre-signed parcel")
+	}
+
+	if len(p.vPkt.Outputs) == 0 {
+		return fmt.Errorf("no outputs in virtual transaction")
+	}
+
+	if p.inputCommitments == nil {
+		return fmt.Errorf("no input commitments in pre-signed parcel")
+	}
+
+	return nil
+}
+
+// PreAnchoredParcel is a request to log and publish an asset transfer of a
+// pre-anchored parcel. All virtual PSBTs and the on-chain BTC level anchor
+// transaction must be fully signed and ready to be broadcast.
+type PreAnchoredParcel struct {
+	*parcelKit
+
+	virtualPackets []*tappsbt.VPacket
+
+	passiveAssets []*tappsbt.VPacket
+
+	anchorTx *tapsend.AnchorTransaction
+}
+
+// A compile-time assertion to ensure PreAnchoredParcel implements the Parcel
+// interface.
+var _ Parcel = (*PreAnchoredParcel)(nil)
+
+// NewPreAnchoredParcel creates a new PreAnchoredParcel.
+func NewPreAnchoredParcel(vPackets []*tappsbt.VPacket,
+	passiveAssets []*tappsbt.VPacket,
+	anchorTx *tapsend.AnchorTransaction) *PreAnchoredParcel {
+
+	return &PreAnchoredParcel{
+		parcelKit: &parcelKit{
+			respChan: make(chan *OutboundParcel, 1),
+			errChan:  make(chan error, 1),
+		},
+		virtualPackets: vPackets,
+		passiveAssets:  passiveAssets,
+		anchorTx:       anchorTx,
+	}
+}
+
+// pkg returns the send package that should be delivered.
+func (p *PreAnchoredParcel) pkg() *sendPackage {
+	log.Infof("New anchored delivery request with %d packets",
+		len(p.virtualPackets))
+
+	// Initialize a package the signed virtual transaction and input
+	// commitment.
+	return &sendPackage{
+		Parcel:         p,
+		SendState:      SendStateLogCommit,
+		VirtualPackets: p.virtualPackets,
+		PassiveAssets:  p.passiveAssets,
+		AnchorTx:       p.anchorTx,
+	}
+}
+
+// kit returns the parcel kit used for delivery.
+func (p *PreAnchoredParcel) kit() *parcelKit {
+	return p.parcelKit
+}
+
+// Validate validates the parcel.
+func (p *PreAnchoredParcel) Validate() error {
```

Comment:
since we expect the signatures to exist, shouldn't we also do some vm/sig checks?

Reply:
if that is not the scope of `Validate` then maybe define what exactly we're validating in the comment above?

Reply:
Good point. Added the comment to the `Validate` method.